src/api-service/__app__/onefuzzlib/tasks/main.py

```diff
@@ -153,6 +153,27 @@ def get_by_task_id(cls, task_id: UUID) -> Union[Error, "Task"]:
         task = tasks[0]
         return task
 
+    @classmethod
+    def get_tasks_by_pool_name(
+        cls, pool_name: str
+    ) -> Optional[Union[Error, List["Task"]]]:
+        tasks = cls.search()
```

Comment:
This will return all tasks regardless of state, then post-filter the results.  `search_states(states=TaskState.available())` will likely provide what you want, which is "all tasks that are not stopping/stopped"

Reply:
Since the result of an empty list and Error is handled the same by the caller, this could be simplified to return List["Task"].